maven/lib/dependabot/maven/file_fetcher.rb

```diff
@@ -11,7 +11,9 @@ class FileFetcher < Dependabot::FileFetchers::Base
                         "profile > modules > module"
 
       def self.required_files_in?(filenames)
-        (%w(pom.xml) - filenames).empty?
+        !filenames.empty? && filenames.all? do |f|
+          f.name.end_with?(".xml") && !f.name.end_with?("pom_parent.xml", "extensions.xml")
+        end
```

Comment:
This one looks incorrect I think, because from the previous implementation I think we should receive an array of `String`, not an array of `DependencyFile`. Can you check? I guess `filenames.any? {|f| f.end_with?(".xml") && f.end_with?("pom_parent.xml", "extensions.xml") }` would do?

If it is indeed incorrect, can we cover the fix with a spec?

Reply:
Thanks for spotting that, @deivid-rodriguez, fixed & added some proper specs for it! And yep, the line you have provided surely worked fine, as we need at least one file to be a proper pom file, and not an extension or a parent. 
One thing i have spotted though... Looks like the maven implementation of the `FileFetcher` don't really invoke this method anywhere, or use it to really validate the fetched files, otherwise the previous implementation would have failed as well, as it required all the files to be only pom.xml, whereas `fetch_files` could already produce and worked with `extensions.xml` dependencies 😃 Guess `required_files_in?` has to be ideally added to the `fetch_files` method, like in some other languages' implementations of `FileFetcher` (but then again, looks like maven is not the only one missing that check 😃 )

Reply:
Nice catch! It seems only Bundler uses this 😬. I guess we can leave that for another time, I opened a ticket about it: https://github.com/dependabot/dependabot-core/issues/6487.

Reply:
Now this comment that you left, @deivid-rodriguez, has really made me think a lot on how that works again, and study a bit deeper the overs logic there, and now i'm ready to discuss that further on!

> I think this PR will fix #5276 too?
Yes and no, at the moment 😃 Will get to this part down the line

> The new behavior seems better since we'll never find any updates for locally pulled dependencies anyways.
Correct, and this is due to now properly searching for the parent files as well, which i have totally forgotten since the time i made it. So my statement in the comment above was a bit incorrect - this PR indeed adds a logic to allow both alternative names for the child modules and for the parent ones, as i recall that fixing just one part seemed not right to me.

> I also looked into why we're using the `pom_parent.xml` trick, and found the PR but not much of an explanation: #356. However, it seems like the current implementation relies on this particular name to be excluded and it would be a non trivial refactoring to get rid of it even if we deemed it as not necessary.
And that is the main question left, if we still need this exclude. I would really assume that for any `pom.xml` file we fetch, no matter parent it is or not - we generally have to check its dependencies and bump them, if possible. 
But now we come to a fun part 😃 Even though i have tried to preserve the non-update-parent logic (as i did not know if that was for a purpose)  - it now allows parent files to slip through it, in case they have custom names and are not named `pom_parent.xml`. I have updated my test repo with some additional pom-s to demonstrate it, so feel free to dry run:
`bin/dry-run.rb maven soulus13/dependabot-maven-test`
There you will see that 2 out of 4 parent poms i have created there have been bumped, and 2 have not been. 

So now it would be great if you could, somehow, dig up why were the parent poms blocked from the dependency checker in the first place, if that is possible, for sure! And anyway, depending on what you think would be the best option - i can either go ahead and completely remove the parent pom exclusion from dependency checking (and only then would the #5276 be fully addressed), or i can adjust a bit the logic within Dependabot::Maven::FileFetcher#recursively_fetch_relative_path_parents to properly substitute the `parent_pom.name` with our "magic file" name, to fall back to the previous behavior for parent poms. My preference would be, for sure, to remove all that "magic" from the code and keep parent poms updated as well, as requested in #5276

> Do you have a particular public repository to run more tests against this PR? It looks mostly ready but I'd like it to give it another pass.
I have been mostly playing with my custom repo above, where i have added all the scenarios i could think of, but here is another real-life sample i have found with custom submodule names: https://github.com/EsotericSoftware/kryo. And for the parent custom module name - https://github.com/filip26/titanium-json-ld is still good enough. Especially if you do locally bulk replace of `pom_parent.xml` to some other magic name, just in case.

Reply:
After the clarification in https://github.com/dependabot/dependabot-core/pull/6248#issuecomment-1398715068, I wonder if (even if currently unused) we should leave this method as is. No strong opinion though.

Reply:
Nope, I do agree that it should be updated to reflect the real logic - one of the files for the dependency inspection must be "pom.xml". As i have force-pushed the branch several times to look nicer after the merge - this thread has now been detached from the source, so i think the current version reflects the reality best:
`filenames.include?("pom.xml")`